skycatalogs/scripts/rotate.py

```diff
@@ -0,0 +1,250 @@
+        c = c/norm
+        s = s/norm
+
+        # nprime = np.array([c*north[0]-s*north[1],
+        #                    s*north[0]+c*north[1]])
+
+        yz_rotation = np.array([[1.0, 0.0, 0.0],
+                                [0.0, c, -s],
+                                [0.0, s, c]])
+
+        second_rotation = np.dot(out_of_self_bases,
+                                 np.dot(yz_rotation,
+                                        to_self_bases))
+
+        self._transformation = np.dot(second_rotation,
+                                      first_rotation)
+
+    def transform(self, ra, dec):
+        """
+        ra, dec are in degrees; return the RA, Dec coordinates
+        of the point about the new field center
+        """
+        xyz = cartesian_from_spherical(np.radians(ra),
+                                       np.radians(dec)).transpose()
+        xyz = np.dot(self._transformation, xyz).transpose()
+        ra_out, dec_out = spherical_from_cartesian(xyz)
+        return np.degrees(ra_out), np.degrees(dec_out)
+
+
+class GalaxyRotator:
+    def __init__(self, field_rotator, field_name, obj_list):
+        self._field_rotator = field_rotator
+        self._field_name = field_name
+        self._obj_list = obj_list
+        self._stride = 1000000
+
+    def output_field_pixels(self, output_dir, arrow_schema):
+        # First make necessary new columns and keep track of output
+        # healpixels
+        field_distinct = set()
+        colls = self._obj_list.get_collections()
+        new_ra = []
+        new_dec = []
+        new_av_ext = []
+        hps = []
+        hps_distinct = []
+        # Make a dict indexed by hp number.  Value is df to be written
+        # to file for that hp
+        hp_out_dict = dict()
+        for c in colls:
+            ra, dec = self._field_rotator.transform(c._ra, c._dec)
+            new_ra.append(ra)
+            new_dec.append(dec)
+            new_av_ext.append(make_MW_extinction_av(ra, dec))
+            rot_hp = np.array(healpy.pixelfunc.ang2pix(NSIDE, ra, dec,
+                                                       lonlat=True))
+            hps.append(rot_hp)
+            distinct = {h for h in rot_hp}
+            hps_distinct.append(sorted(distinct))
+            field_distinct = set.union(field_distinct, distinct)
+
+        # hp_sorted = sorted(field_distinct)
+        native = colls[0].native_columns
+
+        for i, c in enumerate(colls):
+            to_get = [n for n in native if (n != 'ra') and (n != 'dec') and (n != 'MW_av') and n != 'sed_val_bulge' and n != 'sed_val_disk']
+            dat = c.get_native_attributes(to_get)
+            dat['ra'] = new_ra[i]
+            dat['dec'] = new_dec[i]
+            dat['MW_av'] = new_av_ext[i]
+            dat['hp'] = hps[i]
+            # special handling for for tophat SEDs
+            for k in ['sed_val_bulge', 'sed_val_disk', 'sed_val_knots']:
+                dat[k] = c.get_native_attribute(k, no_np=True)
```

Comment:
I've added explanatory comments here and in parquet_reader. The extra argument to get_native_attribute is a complication, but it seemed better to not do something at all rather than do it and then undo it.